fe/fe-common/src/main/java/org/apache/doris/catalog/Type.java

```diff
@@ -2169,6 +2169,17 @@ public static boolean matchExactType(Type type1, Type type2, boolean ignorePreci
                     return false;
                 }
                 return matchExactType(((ArrayType) type2).getItemType(), ((ArrayType) type1).getItemType());
+            } else if (type2.isMapType()) {
```

Comment:
Why we need special branch for decimal,array,map? Can we encapsulate all logic in each type's matchesType() or a new method?